server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -81,22 +81,32 @@ export class AssetRepository implements IAssetRepository {
 
   async getAssetCountByUserId(userId: string): Promise<AssetCountByUserIdResponseDto> {
     // Get asset count by AssetType
-    const res = await this.assetRepository
+    const items = await this.assetRepository
       .createQueryBuilder('asset')
       .select(`COUNT(asset.id)`, 'count')
       .addSelect(`asset.type`, 'type')
       .where('"userId" = :userId', { userId: userId })
       .groupBy('asset.type')
       .getRawMany();
 
-    const assetCountByUserId = new AssetCountByUserIdResponseDto(0, 0);
-    res.map((item) => {
-      if (item.type === 'IMAGE') {
-        assetCountByUserId.photos = item.count;
-      } else if (item.type === 'VIDEO') {
-        assetCountByUserId.videos = item.count;
-      }
-    });
+    const assetCountByUserId = new AssetCountByUserIdResponseDto();
+
+    // asset type to dto property mapping
+    const map: Record<AssetType, keyof AssetCountByUserIdResponseDto> = {
```

Comment:
What is the optimization when changing the previous implementation to this one?

Reply:
In the LivePhotos PR, I added another type of `LIVE_PHOTO`. Would it throw an error if I forget to include the property here?

Reply:
Instead of just returning photos and videos, I think it would be nice if all the types were returned by this endpoint. If you add a new type, like live photos, there would be a type (compilation) error here, like so:

![image](https://user-images.githubusercontent.com/4334196/201344044-0ff4cf67-67fc-4ef3-8a52-03e5832164a2.png)

Reply:
Perfect!